sync command panic

<!--
Please use this template while reporting a bug and provide as much info as possible. Not doing so may result in your bug not being addressed in a timely manner. Thanks!
-->

**What happened**:

./juicefs sync jfs://myfs/random_test1/ jfs://myfs/random_test11/   --worker chaos-k8s-001,chaos-k8s-002,chaos-k8s-003 --list-threads 10 --list-depth 10 --links

```
 Scanned objects: 183292/184050 [==============================================================]  1098.1/s ETA: 0s
Excluded objects: 0                         0.0/s                                      ⠋
 Scanned objects: 184449/186589 [============================================================>-]  1092.2/s ETA: 1s
Excluded objects: 0                         0.0/s                                      ⠸
  Excluded bytes: 0.0 b   (0 Bytes)         0.0 b/s                                    ⠸
 Skipped objects: 0                         0.0/s                                      ⠸
   Skipped bytes: 0.0 b   (0 Bytes)         0.0 b/s                                    ⠸
   Extra objects: 0                         0.0/s                                      ⠸
     Extra bytes: 0.0 b   (0 Bytes)         0.0 b/s                                    ⠸
 Pending objects: 598                       3.5/s                                      ⠸
  Copied objects: 184449                    1092.2/s                                   ⠸
    Copied bytes: 2.0 b   (2 Bytes)         0.0 b/s                                    ⠸
 Checked objects: 184459                    1092.3/s                                   ⠸
   Checked bytes: 3.1 MiB (3245521 Bytes)   18.8 KiB/s                                 ⠸
  Failed objects: 0                         0.0/s                                      ⠸
panic: runtime error: index out of range [0] with length 0

goroutine 210 [running]:
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).doRename.func1(0xc1fee39320)
	/root/chenyunhui/juicefs/pkg/meta/redis.go:1766 +0x271e
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).txn.replaceErrno.func2(0xc00064c620?)
	/root/chenyunhui/juicefs/pkg/meta/redis.go:980 +0x17
github.com/redis/go-redis/v9.(*Client).Watch(0x1?, {0x7f6dac7cc418, 0xc000b185c0}, 0xc1ff422310, {0xc1ff41c740, 0x4, 0x4})
	/root/go/pkg/mod/github.com/redis/go-redis/v9@v9.0.2/tx.go:67 +0xe2
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).txn(0xc000ea62d0, {0x3ebf1a8?, 0xc000b185c0}, 0xc1fb8b7ea0, {0xc1ff41c740?, 0x4, 0x4})
	/root/chenyunhui/juicefs/pkg/meta/redis.go:1013 +0x45c
github.com/juicedata/juicefs/pkg/meta.(*redisMeta).doRename(0xc000ea62d0, {0x3ebf1a8?, 0xc000b185c0}, 0x4480a6d, {0xc200971113, 0x1f}, 0x4480a6d, {0xc20095e9f3, 0x6}, 0x0, ...)
	/root/chenyunhui/juicefs/pkg/meta/redis.go:1695 +0x62a
github.com/juicedata/juicefs/pkg/meta.(*baseMeta).Rename(0xc00007b400, {0x3ebf1a8, 0xc000b185c0}, 0x4480a6d, {0xc200971113, 0x1f}, 0x4480a6d, {0xc20095e9f3, 0x6}, 0x0, ...)
	/root/chenyunhui/juicefs/pkg/meta/base.go:1450 +0x88e
github.com/juicedata/juicefs/pkg/fs.(*FileSystem).Rename(0xc001930460, {0x3ebf1a8?, 0xc000b185c0}, {0xc2009710e0, 0x52}, {0xc20095e9c0, 0x39}, 0x0)
	/root/chenyunhui/juicefs/pkg/fs/fs.go:529 +0x576
github.com/juicedata/juicefs/cmd.(*juiceFS).Put(0xc00193b140, {0xc20095e980?, 0xc00166a1a0?}, {0x3e83240, 0xc2009789d0}, {0x10?, 0x400000000000000?, 0x25555696d569555?})
	/root/chenyunhui/juicefs/cmd/object.go:170 +0x60b
github.com/juicedata/juicefs/pkg/object.(*withPrefix).Put(0xc00193b160, {0xc1fd7ef40d?, 0xc2009746c0?}, {0x3e83240, 0xc2009789d0}, {0x0, 0x0, 0x0})
	/root/chenyunhui/juicefs/pkg/object/prefix.go:118 +0x99
github.com/juicedata/juicefs/pkg/sync.doCopySingle0({0x3ec48d8, 0xc001553b60}, {0x3ec48d8, 0xc00193b160}, {0xc1fd7ef40d, 0x2a}, 0x0, 0x1)
	/root/chenyunhui/juicefs/pkg/sync/sync.go:542 +0x3cd
github.com/juicedata/juicefs/pkg/sync.doCopySingle({0x3ec48d8, 0xc001553b60}, {0x3ec48d8, 0xc00193b160}, {0xc1fd7ef40d, 0x2a}, 0x0, 0x1)
	/root/chenyunhui/juicefs/pkg/sync/sync.go:509 +0x7ae
github.com/juicedata/juicefs/pkg/sync.CopyData.func1()
	/root/chenyunhui/juicefs/pkg/sync/sync.go:774 +0x3d
github.com/juicedata/juicefs/pkg/sync.try(0x3, 0xc146f93a68)
	/root/chenyunhui/juicefs/pkg/sync/sync.go:211 +0xc4
github.com/juicedata/juicefs/pkg/sync.CopyData({0x3ec48d8, 0xc001553b60}, {0x3ec48d8?, 0xc00193b160?}, {0xc1fd7ef40d, 0x2a}, 0x0, 0x1)
	/root/chenyunhui/juicefs/pkg/sync/sync.go:773 +0x177
github.com/juicedata/juicefs/pkg/sync.worker(0xc001616840?, {0x3ec48d8, 0xc001553b60}, {0x3ec48d8, 0xc00193b160?}, 0xc001554000)
	/root/chenyunhui/juicefs/pkg/sync/sync.go:866 +0xe48
github.com/juicedata/juicefs/pkg/sync.Sync.func4()
	/root/chenyunhui/juicefs/pkg/sync/sync.go:1606 +0x65
created by github.com/juicedata/juicefs/pkg/sync.Sync in goroutine 1
	/root/chenyunhui/juicefs/pkg/sync/sync.go:1604 +0x785
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- root@chaos-k8s-001:~/chenyunhui/juicefs# ./juicefs version
juicefs version 1.3.0-dev+2025-03-12.e3e75285
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/redis.go
+++ b/pkg/meta/redis.go
@@ -1472,7 +1472,10 @@ func (m *redisMeta) doUnlink(ctx Context, parent Ino, name string, attr *Attr, s
 		if err := tx.Watch(ctx, m.inodeKey(inode)).Err(); err != nil {
 			return err
 		}
-		rs, _ := tx.MGet(ctx, m.inodeKey(parent), m.inodeKey(inode)).Result()
+		rs, err := tx.MGet(ctx, m.inodeKey(parent), m.inodeKey(inode)).Result()
+		if err != nil {
+			return err
+		}
 		if rs[0] == nil {
 			return redis.Nil
 		}
@@ -1610,7 +1613,10 @@ func (m *redisMeta) doRmdir(ctx Context, parent Ino, name string, pinode *Ino, s
 			return err
 		}
 
-		rs, _ := tx.MGet(ctx, m.inodeKey(parent), m.inodeKey(inode)).Result()
+		rs, err := tx.MGet(ctx, m.inodeKey(parent), m.inodeKey(inode)).Result()
+		if err != nil {
+			return err
+		}
 		if rs[0] == nil {
 			return redis.Nil
 		}
@@ -1768,7 +1774,10 @@ func (m *redisMeta) doRename(ctx Context, parentSrc Ino, nameSrc string, parentD
 		if dino > 0 {
 			keys = append(keys, m.inodeKey(dino))
 		}
-		rs, _ := tx.MGet(ctx, keys...).Result()
+		rs, err := tx.MGet(ctx, keys...).Result()
+		if err != nil {
+			return err
+		}
 		if rs[0] == nil || rs[1] == nil || rs[2] == nil {
 			return redis.Nil
 		}

